src/Playlist.tsx

```diff
@@ -1,8 +1,9 @@
 import React, { useState } from "react";
 import { useHandleFetchAndLoad } from "./useHandleFetchAndLoad";
+import { Songlist } from "./SongList";
 
 export const PlayList: React.FC = () => {
-  const [itemClicked, setItemClicked] = useState(null);
+  const [itemClicked, setItemClicked] = useState<number>(-1);
```

Comment:
Using null here is still right. It seems like we changed this to fix typescript issues; however, the way we'd want to do this is to just update the type. In this case, it would be 
```
useState<number | null>(null)
```

As a rule of thumb, the types you use should NEVER influence the actual code you write. I would sooner want to @ts-ignore before actually changing any logic to adhere to the typing.

Reply:
To elaborate on this, semantically, the state of having not clicking an item is better represented by `null` as opposed to `-1`

Reply:
Finally, the variable name for this is a bit confusing. `itemClicked` is that action taken that changes the state, not the state itself. `itemId, setItemId` would be a more appropriate name here.